### PR TITLE
Task17 implement localisation

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Infrastructure/IlocalizationService.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+
+namespace MoneySplitter.Infrastructure
+{
+    public interface ILocalizationService
+    {
+        IDictionary<string, string> Strings { get; }
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Infrastructure/IlocalizationService.cs]: fixed

File: MoneySplitter.Models/DefinesStringKeys.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace MoneySplitter.Models
+{
+    public enum DefinesStringKeys
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Models/DefinesStringKeys.cs]: i remove this file. File is useless

File: MoneySplitter.Win10/Common/LocalizationServise.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+using System.Xml.Linq;
+using Windows.ApplicationModel;
+using Windows.Storage;
+
+namespace MoneySplitter.Win10.Common
+{
+    public class LocalizationServise : ILocalizationService
+    {
+        public IDictionary<string, string> Strings { get; private set; }
+        public IDictionary<string, string> UdnStrings { get; private set; }
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/Common/LocalizationServise.cs]: fixed

File: MoneySplitter.Win10/Common/LocalizationServise.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                .Descendants()
+                .Elements("data")
+                .ToDictionary(key => key.FirstAttribute.Value, value => (string)value.Element("value"));
+        }
+
+        public string GetValue(string key)
+        {
+            Strings.TryGetValue(key, out string value);
+
+            if (value == "")
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/Common/LocalizationServise.cs]: fixed

File: MoneySplitter.Win10/Common/SearchEngine.cs [hunk 5]
```diff
@@ -88,7 +92,7 @@ public void Activate()
             _previousQuery = string.Empty;
             _query = string.Empty;
             IsStatusMessageVisible = true;
-            StatusMessage = Defines.Search.TEXTBOX_EMPTY;
+            StatusMessage = _localizationService.GetValue("SEURCH_NOT_RESULTS_TEXTBLOCK_TEXT");
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/Common/SearchEngine.cs]: fixed

File: MoneySplitter.Win10/Common/SearchEngine.cs [hunk 6]
```diff
@@ -124,15 +128,15 @@ public void ChangedQeury(string query)
             if (!executionResult.IsSuccess)
             {
                 Results?.Clear();
-                StatusMessage = Defines.ErrorDetails.PROBLEM_SERVER;
+                StatusMessage = _localizationService.GetValue("PROBLEM_SERVER_ERROR");
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/Common/SearchEngine.cs]: fixed

File: MoneySplitter.Win10/CustomControls/CollaboratorControl.xaml [hunk 7]
```diff
@@ -62,16 +63,19 @@
                 Grid.Column="4"
                 Style="{StaticResource CollabaratorButtonStyle}"
                 Visibility="{x:Bind ViewModel.IsGiveButtonVisibil, Mode=OneWay}"
-                Click="OnGiveButtonClick"/>
+                Click="OnGiveButtonClick"
+               extensions:XamlElementExtensions.Resource="REMIND_LOWER_CASE_TEXT_BLOCK_TEXT"/>
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/CollaboratorControl.xaml]: fixed

File: MoneySplitter.Win10/ViewModels/AddTransactionViewModel.cs [hunk 8]
```diff
@@ -21,8 +21,9 @@ public class AddTransactionViewModel : Screen
         private readonly IFilePickerService _filePickerService;
         private readonly INavigationManager _navigationManager;
         private readonly IMembershipService _membershipService;
+        private readonly ILocalizationService _localizationService;
 
-        private string _laberlForTransactionImage = Defines.Register.BrowseImage.AVATAR;
+        private string _laberlForTransactionImage;
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/AddTransactionViewModel.cs]: rename

[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/AddTransactionViewModel.cs]: fixed

File: MoneySplitter.Win10/Styles/GlobalStyles.xaml [hunk 9]
```diff
@@ -37,6 +37,43 @@
         <Setter Property="MaxWidth" Value="{ThemeResource FlyoutThemeMaxWidth}" />
         <Setter Property="MinHeight" Value="{ThemeResource MenuFlyoutThemeMinHeight}"/>
     </Style>
+
+    <Style TargetType="TextBox" x:Key="DefaultTextBoxStyle">
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/Styles/GlobalStyles.xaml]: fixed

File: MoneySplitter.Win10/CustomControls/MobileUserControl.xaml [hunk 10]
```diff
@@ -131,16 +132,23 @@
...
         <Button Click="OnAddOrRemoveButtonClick" 
-                Content="{x:Bind ButtonContent, Mode=OneWay}" 
                 Style="{StaticResource DefaultButtonStyle}"
                 Grid.Column="2"
                 HorizontalAlignment="Right"
-                Visibility="{x:Bind IsButtonVisible, Mode=OneWay}"/>
+                Visibility="{x:Bind IsButtonVisible, Mode=OneWay}">
+            <Button.Content>
+                <Grid>
+                    <TextBlock Text="-"
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/MobileUserControl.xaml]: fixed

File: MoneySplitter.Win10/CustomControls/MobileUserControl.xaml.cs [hunk 11]
```diff
@@ -15,10 +15,10 @@ public UserModel ViewModel
             set { SetValue(ViewModelProperty, value); }
         }
 
-        public string ButtonContent
+        public bool IsRemove
         {
-            get { return (string)GetValue(ButtonContentProperty); }
-            set { SetValue(ButtonContentProperty, value); }
+            get { return (bool)GetValue(IsRemoveProperty); }
```
[the remark this answers is not in the document]
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/MobileUserControl.xaml.cs]: fixed

File: MoneySplitter.Win10/Common/LocalizationServise.cs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+using System.Xml.Linq;
+using Windows.ApplicationModel;
+using Windows.Storage;
+
+namespace MoneySplitter.Win10.Common
+{
+    public class LocalizationServise : ILocalizationService
+    {
+        private IDictionary<string, string> _strings;
+        private IDictionary<string, string> _udnStrings;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/LocalizationServise.cs]: undefined - und

File: MoneySplitter.Win10/Common/LocalizationServise.cs [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+            return
+                XDocument.Parse(xmlRepresentation)
+                .Descendants()
+                .Elements("data")
+                .ToDictionary(key => key.FirstAttribute.Value, value => (string)value.Element("value"));
+        }
+
+        public string GetString(string key)
+        {
+
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/LocalizationServise.cs]: empty line

**Pull request merged**